drivers/can/can_esp32_twai_priv.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Newer ESP32-series MCUs like ESP32-C3 and ESP32-S2 have some slightly different registers
+ * compared to the original ESP32, which is fully compatible with the SJA1000 controller.
+ *
+ * This file adds register definitions for the incompatible registers so that the TWAI frontend
+ * can implement custom implementations (different from sja1000 backend) depending on the MCU.
+ */
+
+#ifndef ZEPHYR_DRIVERS_CAN_ESP32_TWAI_PRIV_H_
+#define ZEPHYR_DRIVERS_CAN_ESP32_TWAI_PRIV_H_
+
+#include "can_sja1000_priv.h"
+
+/*
+ * TWAI_BUS_TIMING_0_REG is incompatible to CAN_SJA1000_BTR0:
+ *   - TWAI_BAUD_PRESC has size of 13 bits instead of 6 bits
+ *   - TWAI_SYNC_JUMP_WIDTH is moved to BIT(14) instead of BIT(6)
+ */
+#define TWAI_BUS_TIMING_0_REG           CAN_SJA1000_BTR0
+
+#ifdef CONFIG_SOC_ESP32
+#define TWAI_BAUD_PRESC_MASK            CAN_SJA1000_BTR0_BRP_MASK
+#define TWAI_SYNC_JUMP_WIDTH_MASK       CAN_SJA1000_BTR0_SJW_MASK
+#else
+#define TWAI_BAUD_PRESC_MASK            GENMASK(12, 0)
+#define TWAI_SYNC_JUMP_WIDTH_MASK       GENMASK(15, 14)
+#endif
+
+#define TWAI_BAUD_PRESC_PREP(brp)	FIELD_PREP(TWAI_BAUD_PRESC_MASK, brp)
+#define TWAI_SYNC_JUMP_WIDTH_PREP(sjw)	FIELD_PREP(TWAI_SYNC_JUMP_WIDTH_MASK, sjw)
+
+/*
+ * TWAI_BUS_TIMING_1_REG is compatible to CAN_SJA1000_BTR1 and only defined for consistency
+ */
+#define TWAI_BUS_TIMING_1_REG           CAN_SJA1000_BTR1
+
+#define TWAI_TIME_SEG1_MASK             CAN_SJA1000_BTR1_TSEG1_MASK
+#define TWAI_TIME_SEG2_MASK             CAN_SJA1000_BTR1_TSEG2_MASK
+#define TWAI_TIME_SAMP                  CAN_SJA1000_BTR1_SAM
+
+#define TWAI_TIME_SEG1_PREP             CAN_SJA1000_BTR1_TSEG1_PREP
+#define TWAI_TIME_SEG2_PREP             CAN_SJA1000_BTR1_TSEG2_PREP
```

Comment:
While it may be more consistent it also makes the code more convoluted. Just use the original SJA1000 definitions where there are no changes.